source/dotnet/Auth/Clouded.Auth.Provider/Controllers/Public/SocialController.cs

```diff
@@ -78,10 +79,20 @@ [Optional] [FromQuery] string? code
 
         if (error != null || code == null)
         {
-            return RedirectPreserveMethod(_socialOptions.Facebook.DeniedRedirectUrl);
+            var deniedUrl = _socialOptions.Facebook.DeniedRedirectUrl;
+            if (!string.IsNullOrEmpty(state))
+            {
+                deniedUrl += deniedUrl.Contains('?') ? $"&state={state}" : $"?state={state}";
```

Comment:
Pre ediaciu URL je lepsie pouzit toto 
https://stackoverflow.com/questions/14517798/append-values-to-query-string